src/util/scroll.ts

```diff
@@ -0,0 +1,34 @@
+import {NgZone} from '@angular/core';
+import {fromEvent, Observable, race} from 'rxjs';
+import {takeUntil, sampleTime} from 'rxjs/operators';
+
+
+const getScrolledParent = (element, includeHidden = false) => {
+  let style = getComputedStyle(element);
+  const excludeStaticParent = style.position === 'absolute';
+  const overflowRegex = includeHidden ? /(auto|scroll|hidden)/ : /(auto|scroll)/;
+
+  if (style.position !== 'fixed') {
+    for (let parent = element; parent = parent.parentElement;) {
+      style = getComputedStyle(parent);
+      if (excludeStaticParent && style.position === 'static') {
+        continue;
+      }
+      if (overflowRegex.test(style.overflow + style.overflowY + style.overflowX)) {
+        return parent;
+      }
+    }
+  }
+
+  return null;
+};
+
+export function ngbScroll(zone: NgZone, hostElement: HTMLElement, callback: () => void, closed$: Observable<any>) {
+  zone.runOutsideAngular(() => {
+    let scrolledParent = getScrolledParent(hostElement);
+    while (scrolledParent) {
+      fromEvent(scrolledParent, 'scroll').pipe(takeUntil(closed$), sampleTime(10)).subscribe(() => { callback(); });
```

Comment:
I think you're using the wrong operator here, should be `throttleTime()` I guess. 
`sampleTime()` emits every 10ms regardless if there was scrolling or not:

![Screen Shot 2019-05-13 at 12 09 13](https://user-images.githubusercontent.com/8074436/57621047-c3a7c700-758a-11e9-984a-d7e13f9db6a4.png)

Reply:
Another comment is that we should probably make `10ms` configurable by the user via the dependency injection, similar to what we're doing for the live announcer.

We can do it separately later of course as things are private for now anyway.